From 9854460c272fc06af53e5db1a9c4d98cf5b231e4 Mon Sep 17 00:00:00 2001
From: Nathan Muggli <nmuggli@google.com>
Date: Thu, 7 May 2026 11:25:20 -0600
Subject: [PATCH] disable boringssl in base::RandBytes and
 NonAllocatingRandomBitGenerator

Introduced in r1045239 (crrev.com/c/3880863).
Updated in r1321254 (crrev.com/c/5669584).
Updated in r1477790 (crrev.com/c/6665689).
Updated in r1624070 (crrev.com/c/7812144).
Updated in r1626560 (crrev.com/c/7828277).
Updated in r1632053 (crrev.com/c/7856143).
Updated in r1682187 (crrev.com/c/8265040).

Also fold in long-term-2000 (disable boringssl used in
base::NonAllocatingRandomBitGenerator in base/rand_util.h) since both
remove <openssl/rand.h> from base/rand_util.

Change-Id: I465dc418929f7ed936d5a2c8be114ca8194acc7e
patch-name: long-term-0800-disable-boringssl-in-base-RandBytes.patch
---
 base/rand_util.h        | 17 -----------------
 base/rand_util_posix.cc | 11 -----------
 2 files changed, 28 deletions(-)

diff --git a/base/rand_util.h b/base/rand_util.h
index ae5425cf17..c08d1e8246 100644
--- a/base/rand_util.h
+++ b/base/rand_util.h
@@ -23,7 +23,6 @@
 #include "base/numerics/safe_conversions.h"
 #include "base/time/time.h"
 #include "build/build_config.h"
-#include "third_party/boringssl/src/include/openssl/rand.h"
 
 namespace memory_simulator {
 class MemoryHolder;
@@ -188,22 +187,6 @@ class RandomBitGenerator {
   ~RandomBitGenerator() = default;
 };
 
-class NonAllocatingRandomBitGenerator {
- public:
-  using result_type = uint64_t;
-  static constexpr result_type min() { return 0; }
-  static constexpr result_type max() { return UINT64_MAX; }
-  result_type operator()() const {
-    uint64_t result;
-    RAND_get_system_entropy_for_custom_prng(reinterpret_cast<uint8_t*>(&result),
-                                            sizeof(result));
-    return result;
-  }
-
-  NonAllocatingRandomBitGenerator() = default;
-  ~NonAllocatingRandomBitGenerator() = default;
-};
-
 // Shuffles [first, last) randomly. Thread-safe.
 template <typename Itr>
 void RandomShuffle(Itr first, Itr last) {
diff --git a/base/rand_util_posix.cc b/base/rand_util_posix.cc
index 6560a29287..597fc33fa3 100644
--- a/base/rand_util_posix.cc
+++ b/base/rand_util_posix.cc
@@ -13,7 +13,6 @@
 #include "base/feature_list.h"
 #include "base/time/time.h"
 #include "build/build_config.h"
-#include "third_party/boringssl/src/include/openssl/rand.h"
 
 #if BUILDFLAG(IS_MAC)
 #include <sys/random.h>
@@ -31,10 +30,6 @@
 #include "base/system/sys_info.h"
 #endif  // BUILDFLAG(IS_MAC)
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
-#include "third_party/lss/linux_syscall_support.h"
-#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
-
 namespace base {
 
 namespace {
@@ -121,12 +116,6 @@ bool UseBoringSSLForRandBytes() {
 namespace {
 
 void RandBytesInternal(span<uint8_t> output, bool avoid_allocation) {
-  // The BoringSSL experiment takes priority over everything else.
-  if (!avoid_allocation && internal::UseBoringSSLForRandBytes()) {
-    // BoringSSL's RAND_bytes always returns 1. Any error aborts the program.
-    (void)RAND_bytes(output.data(), output.size());
-    return;
-  }
 #if BUILDFLAG(IS_MAC)
   CHECK(getentropy(output.data(), output.size()) == 0);
 #elif BUILDFLAG(IS_APPLE)
-- 
2.56.0.rc1.315.gc6ed9934b7-goog

